server/data/data-cassandra/src/main/java/org/apache/james/user/cassandra/CassandraUsersRepository.java

```diff
@@ -67,9 +67,10 @@ public User getUserByName(String name){
         ResultSet result = session.execute(
                 select(REALNAME, PASSWORD, ALGORITHM)
                 .from(TABLE_NAME)
-                .where(eq(REALNAME, name)));
+                .where(eq(NAME, name.toLowerCase())));
         return Optional.ofNullable(result.one())
             .map(row -> new DefaultUser(row.getString(REALNAME), row.getString(PASSWORD), row.getString(ALGORITHM)))
+            .filter(user -> user.getUserName().equals(name))
```

Comment:
No. Name and real name are different (case)

To maintain the previous behaviour (select case sensitive) we need this post filtering.

BTW removing this line makes unit tests fails